packages/compiler/src/render3/r3_template_transform.ts

```diff
@@ -380,36 +370,58 @@ class HtmlAstToIvyAst implements html.Visitor {
         this.bindingParser.parseLiteralAttr(
             name, value, srcSpan, absoluteOffset, attribute.valueSpan, matchableAttributes,
             parsedProperties, keySpan);
+      }
+      return true;
+    }
 
-      } else if (bindParts[IDENT_BANANA_BOX_IDX]) {
-        const keySpan = createKeySpan(srcSpan, '[(', bindParts[IDENT_BANANA_BOX_IDX]);
+    // We didn't see a kw-prefixed property binding, but we have not yet checked
+    // for the []/()/[()] syntax.
+    const DELIMS = {
+      BANANA_BOX: {start: '[(', end: ')]'},
+      PROPERTY: {start: '[', end: ']'},
+      EVENT: {start: '(', end: ')'},
+    };
+    let delims: {start: string, end: string}|undefined;
+    if (name.startsWith(DELIMS.BANANA_BOX.start)) {
+      delims = DELIMS.BANANA_BOX;
+    } else if (name.startsWith(DELIMS.PROPERTY.start)) {
+      delims = DELIMS.PROPERTY;
+    } else if (name.startsWith(DELIMS.EVENT.start)) {
+      delims = DELIMS.EVENT;
+    }
+    if (delims &&
+        // NB: older versions of the parser would match a start/end delimited
+        // binding iff the property name was terminated by the ending delimiter
+        // and the identifier in the binding was non-empty.
```

Comment:
```suggestion
        // Note: older versions of the parser would match a start/end delimited
        // binding iff the property name was terminated by the ending delimiter
        // and the identifier in the binding was non-empty.
```
I admit to having to look up what "NB" stood for. Can we just use "Note:" instead?

Reply:
Yep, done